tests/31sync/08polling.pl

```diff
@@ -0,0 +1,36 @@
+test "That sync can be polled for updates",
+    requires => [qw( first_api_client can_sync )],
+
+    check => sub {
+        my ( $http ) = @_;
+        my ( $user, $filter_id, $room_id, $next );
+
+        matrix_register_user_with_filter( $http, {} )->then( sub {
+            ( $user, $filter_id ) = @_;
+            matrix_create_room( $user )
+        })->then( sub {
+            ( $room_id ) = @_;
+            matrix_sync( $user, filter => $filter_id )
+        })->then( sub {
+            my ( $body ) = @_;
+            $next = $body->{next_batch};
+            Future->wait_any(
+                Future->needs_all(
+                    matrix_sync( $user,
+                        filter => $filter_id, since => $next, timeout => 10000
+                    ),
+                    delay( 0.1 )->then( sub {
+                        matrix_send_room_text_message(
+                            $user, $room_id, body => "1"
+                        )
+                    })
+                ),
+                delay( 10 )->then_fail("Timeout waiting for sync")
```

Comment:
This line isn't necessary; the entire test has a 10 second deadline anyway applied by `run-tests.pl` itself